Automattic/jetpack
# Mobile Stories block (part 6): rewrote StoryEdit class as a function

Title has it - this is just a rewrite.

This PR builds on top of https://github.com/Automattic/jetpack/pull/17539, comes from the suggested comment by @jeherve in https://github.com/Automattic/jetpack/pull/17140#discussion_r505228667.

We're not adding new functionality in this PR, we just rewrote the class as a function. Decided to remove the code that otherwise previously existed in `componentWillUnMount` given the hook referenced there wasn't being really used (and with that, also removed the`doAction`, `hasAction` imports).

To test:
Test with the PRs related to the base branch, as follows (but updating the jetpack branch pointed to by the gutenberg-mobile submodule to this one)
Gutenberg-mobile wordpress-mobile/gutenberg-mobile#2747
WPANdroid wordpress-mobile/WordPress-Android#13174
1. run the host app
2. verify the blocks render correcly, upload and save progress is correctly represented, etc.
